More use_grabs, standardizes checks

emmanuelbassil opened the pull request:
🆑 emmanuelbassil
bugfix: Grabbing someone that is weakly grabbed by someone else and walking away should now release the other person's grab. No more infinite range grabs.
bugfix: Fixed not being able to put mobs into suit cyclers. You require a weak grab to put a mob into a suit cycler, suit storage, or gibber. Can click and drag self.
/🆑 


I did a few things:

1. Moved the sanity check for moving a mob inside a machine at the level of obj/machinery to minimize copy pasting.
2. Added use_grab for the suit cyclers and the gibber, also allowed to use mouse to click-drag self (and only self) into then.
3. Removed ability to click drag a mob into gibber. Existing code wanted you to have an upgraded grab or just click drag the person. Can still click drag self.
4. Changes a bunch of qdel(grabs) to either using let_go() (which exists to allow custom flavor for releasing grabs, unused atm) or remove_grabs_and_pulls; which removes all grabs and pulls affecting a mob depending on context. I left qdel(grabs) in cases where the grab is used in combat like table smashes, where you don't want all grabs removed and you don't want custom grab release flavor.
5. Added a sanity check under grab process(); because grabbing someone and dragging him away does not remove the grab of other users on said person allowing an infinite range grab. If this is too intensive, please let me know.

emmanuelbassil commented:
Tested putting in mobs/self into all of the above machines and it works fine. 
Ready for review

emmanuelbassil commented:
Fixed if spacing
Added forgotten type check for mousedrop in suit cycler, storage, and gibber.
Ready again!

juxjux9930 commented on code/game/machinery/_machines_base/machinery.dm:
```diff
@@ -259,6 +258,43 @@
+	if (panel_open)
+		to_chat(user, SPAN_WARNING("Close the maintenance panel before attempting to place [user == target ? "yourself" : "\the [target]"] in \the [src]."))
+		return FALSE
+	for (var/mob/living/carbon/slime/slime in range(1,target))
```
```suggestion
	for (var/mob/living/carbon/slime/slime in range(0,target))
```
`range(1,target)` is checking the target's turf and all neighboring turfs, but if the slime is glomping on you it's going to be on your turf.

juxjux9930 commented on code/game/machinery/_machines_base/machinery.dm:
```diff
@@ -259,6 +258,43 @@
+		return FALSE
+	for (var/mob/living/carbon/slime/slime in range(1,target))
+		if (slime.Victim == target)
+			to_chat(user, "[target] will not fit into \the [src] because they have a slime latched onto their head.")
```
```suggestion
			to_chat(user, "[target] will not fit into \the [src] because a slime has latched onto them!")
```
Slimes are many things, but I don't think they jump onto people's heads- might be wrong though.

juxjux9930 commented on code/game/machinery/_machines_base/machinery.dm:
```diff
@@ -259,6 +258,43 @@
+ * Call parent first if you want to add new checks specific for each machine, this proc handles the common stuff.
+ * Returns TRUE if mob can be moved into a machine, FALSE if it cannot.
+ */
+/obj/machinery/proc/user_can_move_target_inside(mob/target, mob/user)
```
If every override of this contains/is *just*
 ```
 if (occupant)
	to_chat(user, SPAN_WARNING("\The [src] is already occupied!"))
	return FALSE
```
Could/Should we move the occupant var to `/obj/machinery`, or maybe make a shared parent for machinery that can have occupants?

juxjux9930 commented on code/game/machinery/suit_storage.dm:
```diff
@@ -157,6 +129,49 @@
 	SSnano.update_uis(src)
 
+/obj/machinery/suit_storage_unit/proc/move_target_inside(mob/target, mob/user)
+	visible_message(SPAN_WARNING("\The [user] starts putting \the [target] into the Suit Storage Unit."))
```
```suggestion
	visible_message(SPAN_WARNING("\The [user] starts putting \the [target] into \the [src]."))
```

juxjux9930 commented on code/game/machinery/suit_storage.dm:
```diff
@@ -157,6 +129,49 @@
+	if (!isopen)
+		to_chat(user, SPAN_NOTICE("The unit's doors are shut."))
+		return FALSE
+	if (occupant || helmet || suit || boots || tank || mask)
```
```suggestion
	if (occupant || suit || tank || (helmet && boots && mask))
```
Shoes, a helmet and a facemask probably aren't large enough to prevent you shoving a person inside individually, but together they might be

emmanuelbassil commented:
Handled!
You are right that slimes glomp the entire being, it's been a while since I saw one run amock.

emmanuelbassil committed:
More use_grabs, standardizes checks


Fixes


Auto stash before rebase of "refs/heads/Fixes"
Auto stash before rebase of "refs/heads/Fixes"
Fixes


Auto stash before rebase of "refs/heads/Fixes"
Auto stash before rebase of "refs/heads/grabs"
Auto stash before rebase of "refs/heads/grabs"

emmanuelbassil commented:
I *THINK* I properly pulled dev and rebased.
So this should be ready for review again.

emmanuelbassil commented:
Fixed merge  conflict
